From: Michal Hocko <mhocko@suse.cz>
Subject: kabi, unix: properly account for FDs passed over unix sockets
Patch-mainline: never, kabi fixup
References: bnc#839104

move the newly added unix_inflight to the end of the structure and hide it
from the kabi checker. Nobody should allocate user_struct (this is a deeply
internal data structure) and nobody outside of the unix socket proper should
ever touch it.

Signed-off-by: Michal Hocko <mhocko@suse.cz>

---
 include/linux/sched.h |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/include/linux/sched.h
+++ b/include/linux/sched.h
@@ -710,7 +710,6 @@ struct user_struct {
 	unsigned long mq_bytes;	/* How many bytes can be allocated to mqueue? */
 #endif
 	unsigned long locked_shm; /* How many pages of mlocked shm ? */
-	unsigned long unix_inflight;	/* How many files in flight in unix sockets */
 
 #ifdef CONFIG_KEYS
 	struct key *uid_keyring;	/* UID specific keyring */
@@ -727,6 +726,7 @@ struct user_struct {
 #endif
 #ifndef __GENKSYMS__
 	atomic_long_t pipe_bufs;  /* how many pages are allocated in pipe buffers */
+	unsigned long unix_inflight;	/* How many files in flight in unix sockets */
 #endif
 };
 
